middle_end/flambda/inlining/inlining_decision.ml

```diff
@@ -134,6 +149,7 @@ module Call_site_decision = struct
     | Max_inlining_depth_exceeded
     | Recursion_depth_exceeded
     | Never_inline_attribute
+    | Rejected_by_cost_metrics
     | Inline of {
```

Comment:
This case should probably be rewritten to better distinguish the reasons why we can inline a function call:
- an attribute forced the inlining
- the function was marked as "small" at declaration time and thus unconditionally inlined
- the speculative inlining returned a positive result (in which case the result of specualitve inlining should be recorded in the decision)

Reply:
I added more variants to describe the different decisions a bit better.